src/metabase/query_processor/middleware/mark_outer_fields.clj

```diff
@@ -0,0 +1,36 @@
+(ns metabase.query-processor.middleware.mark-outer-fields
+  (:require [clojure.walk :as walk]
+            [metabase.mbql.util :as mbql.u]
+            [metabase.query-processor.util :as qp.util]))
+
+(defn- matching-fields?
+  [field source-field]
+  (= (qp.util/field-ref->key field) (qp.util/field-ref->key source-field)))
```

Comment:
this expects a field_ref, not a field. You'll need `(comp qp.util/field-ref->key :field_ref)` as the function applied before checking equality.

Reply:
I'm confused. This is supposed to be called with two triples (vectors of length 3) with the first value being `:field`, the second an id or a name and we don't care about the third. Calling `:field_ref` would return `nil`. I'll have a look to see where `:field_ref`s come into play.